src/java/org/apache/cassandra/index/sai/plan/Plan.java

```diff
@@ -328,6 +323,17 @@ protected String description()
         return "";
     }
 
+    /**
+     * Returns the index context if the plan node uses one.
+     * Need to be overridden by nodes that use an index.
+     * None-recursive. Use {@link #forEach(Function)} to get the index context of all nodes.
+     */
+    protected @Nullable IndexContext getIndexIfPresent()
```

Comment:
you don't need to append `ifPresent` to the name. It is obvious that if the method returns `@Nullable` then it signals "no index" by returning null.

Reply:
"None-recursive." -> "Non-recursive"

Reply:
> Use {@link #forEach(Function)} to get the index context of all nodes.

That is misleading. It suggests there is a single context of all nodes, when in fact each node can refer to a different IndexContext. I'd just remove the whole line. It is enough to say it returns the index referenced by this particular node only.

Reply:
Addressed all three suggestions